bundles/binding/org.openhab.binding.fatekplc/src/main/java/org/openhab/binding/fatekplc/internal/FatekPLCBinding.java

```diff
@@ -0,0 +1,278 @@
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Fatek PLC Refresh Service";
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the configuration is removed or mandatory references
+	 * are no longer satisfied or the component has simply been stopped.
+	 *
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 Unspecified
+	 *            <li>1 The component was disabled
+	 *            <li>2 A reference became unsatisfied
+	 *            <li>3 A configuration was changed
+	 *            <li>4 A configuration was deleted
+	 *            <li>5 The component was disposed
+	 *            <li>6 The bundle was stopped
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+
+		setProperlyConfigured(false);
+		logger.info("deactivate executed reason={}", reason);
+	}
+
+	@Override
+	public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
+
+		logger.debug("update start");
+		try {
+			setProperlyConfigured(false);
+
+			// stop and clear current slaves
+			for (FatekPLCSlave slave : slaves.values()) {
+				try {
+					slave.disconnect();
+				} catch (FatekIOException e) {
+					logger.warn("diconnect", e);
+				}
+			}
+			slaves.clear();
+
+			// global configurations
+			refreshInterval = getLongProperty(properties, "refresh", refreshInterval);
+
+			// slaves configurations
+			Enumeration<String> keys = properties.keys();
+			while (keys.hasMoreElements()) {
+
+				String key = keys.nextElement();
+				// skip global configs item
+				if (GLOBAL_CONFIG_NAMES.contains(key)) {
+					continue;
+				}
+
+				String keyItems[] = key.split("\\.");
+				if (keyItems.length != 2) {
+					throw new ConfigurationException(key, "incorrect format, should be slave.property");
+				}
+
+				FatekPLCSlave slave = slaves.get(keyItems[0]);
+				if (slave == null) {
+					slave = new FatekPLCSlave(keyItems[0], eventPublisher, itemRegistry);
+					slaves.put(slave.getName(), slave);
+				}
+
+				slave.configure(keyItems[1], properties.get(key));
+			}
+
+		} catch (ConfigurationException e) {
+			logger.error("", e);
+			throw e;
+		}
+
+		setProperlyConfigured(true);
+		logger.info("Configuration has been updated, refresh={}", refreshInterval);
+	}
+
+	/**
+	 * Parse long from property with default.
+	 *
+	 * @param properties
+	 *            properties map
+	 * @param key
+	 *            given key
+	 * @param defaultVale
+	 * @return long value for given key
+	 */
+	private long getLongProperty(Dictionary<String, ?> properties, String key, long defaultVale) {
+
+		String value = (String) properties.get(key);
+		long ret;
+		try {
+			ret = Long.parseLong(value);
+		} catch (NumberFormatException e) {
+			ret = defaultVale;
+			logger.warn("Property {} parse exception {}", key, e.getMessage());
+		}
+
+		return ret;
+	}
+
+	/**
+	 * @{inheritDoc}
```

Comment:
Binding skeleton was generated by maven archetype as I remember.
So this was wrong in code of archetype, we can find many such misspell in other bindings.
Of course I will correct all occurrence in my code.

If you want I can prepare new PR which correct this in other bindings.